Introduce a docker-compose based testing method (#4121)
perma-id/w3id.org: README.md

URL that you want to redirect to, and the HTTP code that you want to use
when redirecting. An administrator will then create the redirect for you.

#### Testing Your Changes

In order to make sure your changes are valid and work correctly, a
[`docker-compose.yml`](./docker-compose.yml) file is available at the root
of this project to be able to run a server on your local machine for
testing purposes.

Having a working `docker-compose` install is necessary, please refer
to the [official Docker documentation](https://docs.docker.com/compose/install/)
about this for this prerequisite.

Once you are ready, you can run the following command to bring up your
local testing server.

```bash
docker-compose up -d
```

This will start an Apache httpd server with [`httpd.conf`](./httpd.conf)
as a configuration file.

You can now test out your identifier by replacing `https://w3id.org/` by
`http://localhost:8080/` in your query, for example:

```
https://w3id.org/identifier/resource
```

> Will become:

Reviewer: ```suggestion
— will become —
```

```
http://localhost:8080/identifier/resource
```

> Don't hesitate to check out the container's logs with the following
> command as the Apache httpd configuration has been tweaked for a high
> level of debugging information on the `rewrite_module`.

Reviewer: ```suggestion
Check the container's logs with the following
command. The Apache `httpd` configuration has been tweaked to deliver a high
level of debugging information for the `rewrite_module`.
```

```bash
docker-compose logs -f
```

> When your testing is finished, all you need to do is to shutdown your

Reviewer: ```suggestion
When your testing is finished, all you need to do is to shut down your
```

> server by running.

Reviewer: ```suggestion
server by running this command:
```
